### PR TITLE
[Feature] Build ICU Daily Round Screen

### LINKED ISSUE #42: Build out the ICU Daily Round Screen
From the main menu there is a button which takes you to the ICU General Care screen.  That screen is currently **really** basic.

**We need to build build out a detailed version of the  ICU General Care screen**

@greggmiller Can you please provide some information in this issue on what should be in this section?  Once that is done assign the issue to Marc Edwards.

@marcedwards Once greg has provided info, can you provide some design guidance in this issue in on how it should look.  Once this is done, can you label this issue as "Ready for Dev" so that one of the devs can pick it up.

### PR DESCRIPTION
**Pull request by yimingc2**
resolved #42

## Abstract
This PR will add ICU Daily Round Screen.

- [x] `release-notes.txt` has been updated.
(👆 delete this line if your changes have no user facing features)

## Changes

- Create models for daily round steps and checklist
- Add hard data to feed the screen
- Replace `general_care_view` with `daily_round_view`
- Extracted out checklist widget for reuse

## Screenshots
Simulator Screen Shot - iPhone 11 Pro - 2020-04-07 at 19 09 34
Simulator Screen Shot - iPhone 11 Pro - 2020-04-07 at 19 09 39

## Things to note
There are 2 things that I haven't implemented/solved in this PR. Needs more discussion as well.
- Part of the text needs to be bold.
- Some icons can not be displayed properly because of the font.

### CODE REVIEW

**Review by workerbee22 — APPROVED** [review]
LGTM

File: lib/widget/containers/icu_daily_round_container.dart [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  }
+}
+
+// Steps Container
+class ICUDailyRoundStepsContainer extends StatelessWidget {
+  final List<ICUDailyRoundStepSection> sections;
+
+  const ICUDailyRoundStepsContainer({this.sections});
+
+  List<Widget> __renderList() {
```
Comment (maks) [inline lib/widget/containers/icu_daily_round_container.dart]: one underscore in the name is the usual naming convention

**Review by maks — CHANGES_REQUESTED** [review]
:tada: Great work!
Just one small thing and please see if you are able to use Provider for the checkbox state

**Comment by yimingc2** [comment]
> 🎉 Great work!
> Just one small thing and please see if you are able to use Provider for the checkbox state

Yeah, will review Bramley's job.

**Comment by lukesleeman** [comment]
Unfortunately I am no flutter developer, so I can't do much to help out - 🤞 the following information is helpful:

I tested out the persistence of the intubation checklist items in this branch, and yes - it doesn't appear to be working.  I _think_ this has something to do with the changes to `intubation_checklist_item.dart` - we can see that `import 'package:provider/provider.dart';` was removed for example.  But unfortunately beyond that I can't help figure out whats going on 😕

**Comment by yimingc2** [comment]
> Unfortunately I am no flutter developer, so I can't do much to help out - 🤞 the following information is helpful:
> 
> I tested out the persistence of the intubation checklist items in this branch, and yes - it doesn't appear to be working. I _think_ this has something to do with the changes to `intubation_checklist_item.dart` - we can see that `import 'package:provider/provider.dart';` was removed for example. But unfortunately beyond that I can't help figure out whats going on 😕

Hey Luke, I think it's a rebase issue. I didn't know someone else also worked on that file. Just fixed it and pushed. Now both checklist can save the checked status.

File: lib/view/airway/checklist/intubation_checklist_page.dart [hunk 2]
```diff
@@ -47,7 +28,7 @@ class IntubationChecklistPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return ChangeNotifierProvider(
-        create: (context) => IntubationChecklistSelectionProvider(),
+        create: (context) => ChecklistSelectionProvider<IntubationChecklistItem>(),
```
Comment (maks) [inline lib/view/airway/checklist/intubation_checklist_page.dart]: :100: :+1: for making it a generic ChecklistSelectionProvider

File: lib/widget/containers/intubation_checklist_container.dart [hunk 3]
```diff
@@ -28,7 +30,7 @@ class IntubationChecklistContainer extends StatelessWidget {
           child: ListView(
             children: [
               Container(padding: const EdgeInsetsDirectional.only(top: 10)),
-              ...renderList(),
+              ...__renderList(),
```
Comment (maks) [inline lib/widget/containers/intubation_checklist_container.dart]: just one underline in the name please :pray: :slightly_smiling_face:

**Review by maks — APPROVED** [review]
:tada: simply excellent work @yimingc2 !

File: lib/widget/containers/intubation_checklist_container.dart (hunk 3 above)
Comment (yimingc2) [reply lib/widget/containers/intubation_checklist_container.dart]: Just removed. Also changed the reusable checklist widget from Stateful to be Stateless due to the use of provider. Learnt a lot from it. Thanks for the guide!

**Pull request merged**